File: support_tools/swigwin-3.0.12/Examples/test-suite/javascript/typedef_inherit_runme.js
var typedef_inherit = require("typedef_inherit");

a = new typedef_inherit.Foo();
b = new typedef_inherit.Bar();

x = typedef_inherit.do_blah(a);
if (x != "Foo::blah")
    print("Whoa! Bad return" + x);

x = typedef_inherit.do_blah(b);
if (x != "Bar::blah")
    print("Whoa! Bad return" + x);

c = new typedef_inherit.Spam();
d = new typedef_inherit.Grok();

x = typedef_inherit.do_blah2(c);
if (x != "Spam::blah")
    print("Whoa! Bad return" + x);

x = typedef_inherit.do_blah2(d);
if (x != "Grok::blah")
    print ("Whoa! Bad return" + x);

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/typedef_inherit_runme.py
import typedef_inherit

a = typedef_inherit.Foo()
b = typedef_inherit.Bar()

x = typedef_inherit.do_blah(a)
if x != "Foo::blah":
    print "Whoa! Bad return", x

x = typedef_inherit.do_blah(b)
if x != "Bar::blah":
    print "Whoa! Bad return", x

c = typedef_inherit.Spam()
d = typedef_inherit.Grok()

x = typedef_inherit.do_blah2(c)
if x != "Spam::blah":
    print "Whoa! Bad return", x

x = typedef_inherit.do_blah2(d)
if x != "Grok::blah":
    print "Whoa! Bad return", x
